app/herdbook.py

```diff
@@ -474,6 +480,304 @@ def get_mean_kinship(g_id):
     return {}
 
 
+@APP.route("/api/certificates/update/<i_number>", methods=["PATCH"])
+@login_required
+def update_certificate(i_number):
+    """
+    Returns an updated pdf of the individual given by `i_number`.
+    """
+    user_id = session.get("user_id", None)
+    ind = da.get_individual(i_number, user_id)
+
+    if ind is None:
+        return jsonify({"response": "Individual not found"}), 404
+
+    certificate_exists = ind.get("certificate", None)
+    uploaded = False
+
+    try:
+        present = check_certificate_s3(ind_number=ind["number"])
+        if certificate_exists and present:
+            data = get_certificate_data(ind, user_id)
+            data.update(**request.json)
+            pdf_bytes = get_certificate(data)
+            signed_data = sign_data(pdf_bytes)
+            uploaded = upload_certificate(
+                pdf_bytes=signed_data.getvalue(), ind_number=ind["number"]
+            )
+    except Exception as ex:  # pylint: disable=broad-except
+        APP.logger.info("Unexpected error while updating certificate " + str(ex))
+        return jsonify({"response": "Error processing your request"}), 404
+
+    if uploaded:
+        return create_pdf_response(
+            pdf_bytes=signed_data, obj_name=f'{ind["certificate"]}.pdf'
+        )
+
+    return jsonify({"response": "Certificate was not updated"}), 404
+
+
+@APP.route("/api/certificates/issue/<i_number>", methods=["POST"])
+@login_required
+def issue_certificate(i_number):
+    """
+    Returns an issued pdf certificate of the individual given by `i_number`.
+    """
+    user_id = session.get("user_id", None)
+    ind = da.get_individual(i_number, user_id)
+    if ind is None:
+        return jsonify({"response": "Individual not found"}), 404
+    certificate_exists = ind.get("certificate", None)
+    if certificate_exists:
+        return jsonify({"response": "Certificate already exists"}), 400
+
+    ##cert_number = str(random.randint(50001, 99999))
+    cert_number = "99999"
+    ind.update(**request.json, certificate=cert_number)
+    da.update_individual(ind, session.get("user_id", None))
```

Comment:
I don't understand how this can work as is, and also think it's not the right way to do it.

```suggestion
    ind.certificate=cert_number
    ind.save()
```

? 

I've seen https://docs.peewee-orm.com/en/latest/peewee/querying.html#atomic-updates but do not think it applies here.